### ISSUE TITLE
Move comparisons for equality of value types before equality comparisons of collections

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#1090]
We should report that equality comparisons within logical AND conditions should be done by comparing values first and collections last.

Example:
```c#
public class SomeData
{
    public int Id { get; set; }
    public List<Data> Data { get; set; }

    public bool Equals(SomeData other)
    {
        return ((Data == other.Data) || (Data != null && Data.SequenceEqual(otherData))) && Id == other.Id;
    }
}
```

should be reported and changed into:
```c#
public class SomeData
{
    public int Id { get; set; }
    public List<Data> Data { get; set; }

    public bool Equals(SomeData other)
    {
        return Id == other.Id && ((Data == other.Data) || (Data != null && Data.SequenceEqual(otherData)));
    }
}
```

The reason is that comparison on value types is normally faster than comparing lists.
